File: javascript_basics/src/tests/navigation.test.js
require("../testSetup.js");
const { By } = require("selenium-webdriver");
const { getDriver } = require("../globalSetup.js");
const { logger } = require("../utils/logger.js");
const { sleep } = require("../utils/sleep.js");

describe("Navigation & Page Title Validation", function () {
  it("should navigate to the Form Authentication page and verify header", async function () {
    logger.info("STARTING Test --- Navigation & Page Title Validation ---");

    const url = "https://the-internet.herokuapp.com/";
    logger.info(`Navigating to ${url}`);
    await getDriver().get(url);
    // await sleep(1000);

    logger.info("Locating 'Form Authentication' link");
    const link = await getDriver().findElement(By.linkText("Form Authentication"));

    logger.info("Clicking 'Form Authentication' link");
    await link.click();
    // await sleep(1000);

    logger.info("Validating header equals 'Login Page'");
    const header = await getDriver().findElement(By.tagName("h2"));
    const headerText = await header.getText();
    // await sleep(1000);

    if (headerText !== "Login Page") {
      throw new Error(`Unexpected header text: ${headerText}`);
    }

    const currentUrl = await getDriver().getCurrentUrl();
    logger.info(`Current URL: ${currentUrl}`);

    if (!currentUrl.includes("login")) {
      throw new Error(`Expected URL to contain 'login', but got: ${currentUrl}`);
    }

    logger.info("ENDING TEST --- Navigation & Page Title Validation ---");
  });
});

File: csharp_basics/tests/navigation_test.cs
using NUnit.Framework;
using OpenQA.Selenium;
using csharp_basics.utils;

namespace csharp_basics.tests
{
    [TestFixture]
    public class NavigationTest : BaseTest
    {
        [Test]
        public void TestNavigationAndTitle()
        {
            Logger.Write("STARTING Test --- Navigation & Page Title Validation ---");

            // Step 1: Navigate to main page
            string url = "https://the-internet.herokuapp.com/";
            Logger.Write($"Navigating to {url}");
            driver.Navigate().GoToUrl(url);

            // Step 2: Locate and click the 'Form Authentication' link
            Logger.Write("Locating 'Form Authentication' link");
            IWebElement link = driver.FindElement(By.LinkText("Form Authentication"));
            Logger.Write("Clicking 'Form Authentication' link");
            link.Click();

            // Step 3: Validate that navigation succeeded
            Logger.Write("Validating that header text equals 'Login Page'");
            string header = driver.FindElement(By.TagName("h2")).Text;
            Assert.That(header, Is.EqualTo("Login Page"));

            // Step 4: Confirm URL contains "login"
            string currentUrl = driver.Url;
            Logger.Write($"Current URL after navigation: {currentUrl}");
            Assert.That(currentUrl, Does.Contain("login"));

            Logger.Write("ENDING TEST --- Completed successfully: Navigation & Page Title Validation ---");
        }
    }
}
